lib/ui/shared/products_grid.dart

```diff
@@ -3,14 +3,32 @@ import 'package:flutter/material.dart';
 class ProductsGrid extends StatefulWidget {
   final String productName;
   final String price;
+  final String image;
   const ProductsGrid(
-      {super.key, required this.productName, required this.price});
+      {super.key,
+      required this.productName,
+      required this.price,
+      required this.image});
 
   @override
   State<ProductsGrid> createState() => _ProductsGridState();
 }
 
 class _ProductsGridState extends State<ProductsGrid> {
+  bool isClicked = false;
+
+  void saveItem() async {
+    if (isClicked) {
```

Comment:
You don't have to call setState twice.
This logic can be simplified to 

```dart
void saveItem() {
setState(() { isClicked = !isClicked; });
}
```

That's sufficient because you're only toggling the value of `isClicked`. "!", which is binary NOT, does that toggling for you. Also, I'm not sure why your marked the method as `async` but that is not necessary since you're not executing any asynchronous task in the method.